src/sentry/profiles/consumers/process/factory.py

```diff
@@ -1,19 +1,30 @@
 from typing import Mapping
 
 from arroyo.backends.kafka.consumer import KafkaPayload
-from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.abstract import (
+    MessageRejected,
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
 from arroyo.processing.strategies.commit import CommitOffsets
 from arroyo.processing.strategies.run_task import RunTask
 from arroyo.types import Commit, Message, Partition
 
+from sentry.monitoring.queues import is_queue_healthy, monitor_queues
 from sentry.profiles.task import process_profile_task
 
 
 def process_message(message: Message[KafkaPayload]) -> None:
+    if not is_queue_healthy("profiles.process"):
+        raise MessageRejected()
     process_profile_task.s(payload=message.payload.value).apply_async()
 
 
 class ProcessProfileStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self) -> None:
+        super().__init__()
+        monitor_queues()
```

Comment:
@phacops and I just talked about this. We're aware that different consumers each get their own instance of the monitoring process and think it's fine that they all write to Redis, since they all get the same data from RabbitMQ.